### PR TITLE
Introduce 'tilt' frame

### PR DESCRIPTION
**Pull request by kbwestfall**
Resubmitting:

This introduces the 'tilt' frame type, but it's not currently used by anything. When typing frames, anything typed as an 'arc' is also now typed as a 'tilt'.

### CODE REVIEW

File: pypeit/core/framematch.py [hunk 1]
```diff
@@ -91,365 +92,95 @@ def check_frame_exptime(exptime, exprng):
...
-            msgs.warn('No {0} frames for {1}'.format(ftag, target)
-                      + ', but will use MasterFrames.')
-        else:
-            insufficient_frame_error(ftag)
-
-    return code
 
 
+# TODO: May want to keep this in case we ever try to bring it back....
+#def group_AB_frames(file_list, targets, coords, max_nod_sep=2):
```
Comment (profxj) [inline pypeit/core/framematch.py]: what do you think @tiffanyhsyu ?

File: pypeit/par/pypeitpar.py [hunk 2]
```diff
@@ -164,8 +167,9 @@ def valid_frame_types():
         """
         Return the list of valid frame types.
         """
-        return [ 'bias', 'dark', 'pixelflat', 'arc', 'pinhole', 'trace', 'standard', 'science',
-                 'all' ]
+        return FrameTypeBitMask().keys()
+#        return [ 'bias', 'dark', 'pixelflat', 'arc', 'pinhole', 'trace', 'standard', 'science',
```
Comment (profxj) [inline pypeit/par/pypeitpar.py]: remove

File: pypeit/par/pypeitpar.py [hunk 3]
```diff
@@ -1652,213 +1659,7 @@ def validate(self):
...
-        """
-        Return the list of valid functions to use for object tracing.
-        """
-        return [ 'gaussian', 'gaussfunc', 'moffat', 'moffatfunc' ]
-
-    def validate(self):
-        pass
-
-# ToDO place holder to be updated by JFH
 class ScienceImagePar(ParSet):
```
Comment (profxj) [inline pypeit/par/pypeitpar.py]: this is probably not the time to do this, but I should love for the Class names here to be identical to their names in the parset.  Just sayin'

File: pypeit/spectrographs/gemini_gnirs.py [hunk 4]
```diff
@@ -176,7 +176,7 @@ def check_frame_type(self, ftype, fitstbl, exprng=None):
         if ftype == 'pinhole' or ftype == 'dark' or ftype == 'bias':
             # Don't type pinhole, dark, or bias frames
             return np.zeros(len(fitstbl), dtype=bool)
-        if ftype == 'arc':
+        if ftype == 'arc' or ftype == 'tilt':
```
Comment (profxj) [inline pypeit/spectrographs/gemini_gnirs.py]: FWIW, I prefer:

```
if ftype in ['arc', 'tilt']
```

makes it easy to keep adding.

File: pypeit/spectrographs/keck_nires.py [hunk 5]
```diff
@@ -171,6 +154,7 @@ def check_frame_type(self, ftype, fitstbl, exprng=None):
         """
         Check for frames of the provided type.
         """
+        # TODO: Arcs, tilts, darks?
```
Comment (profxj) [inline pypeit/spectrographs/keck_nires.py]: not sure.  Pinging @jhennawi

File: pypeit/par/pypeitpar.py (hunk 2 above)
Comment (kbwestfall) [reply pypeit/par/pypeitpar.py]: Done.

File: pypeit/par/pypeitpar.py (hunk 3 above)
Comment (kbwestfall) [reply pypeit/par/pypeitpar.py]: You mean you want this to be called `ReducePar` to match that these are sent to the `Reduce` class? If so, then I have more comments/questions.  But yes, I think we should add this as an issue for a later PR.

File: pypeit/spectrographs/gemini_gnirs.py (hunk 4 above)
Comment (kbwestfall) [reply pypeit/spectrographs/gemini_gnirs.py]: Agreed.  Changed.  Although I (irrationally?) dislike, e.g., `if ftype in ['arc']:`

File: pypeit/par/pypeitpar.py (hunk 3 above)
Comment (jhennawi) [reply pypeit/par/pypeitpar.py]: I agree with this point. However there is something else called reduce I think, i.e. the top-level parset.

File: pypeit/spectrographs/keck_nires.py (hunk 5 above)
Comment (jhennawi) [inline pypeit/spectrographs/keck_nires.py]: This is not my TODO, and I've never altered init_meta.

File: pypeit/par/pypeitpar.py (hunk 3 above)
Comment (kbwestfall) [reply pypeit/par/pypeitpar.py]: Let's add this as a topic for the next telecon.

File: pypeit/spectrographs/keck_nires.py (hunk 5 above)
Comment (kbwestfall) [reply pypeit/spectrographs/keck_nires.py]: No, it was mine.  The point of the TODO was to note that this spectrograph doesn't type these frames.  If that doesn't matter, we can move on, but we should note this in the docstring.  I'd like to just leave it as a TODO for now.

**Comment by kbwestfall** [comment]
Tests pass.  Merging.  I'll run the Dev suite on the develop branch tonight.

**Pull request merged**